back/configuraBack.txt
## CONFIGURA BACK 
### https://developer.mozilla.org/es/docs/Learn/Server-side/Django/Models
### https://docs.djangoproject.com/es/3.2/ref/models/querysets/
### https://docs.djangoproject.com/en/3.2/ref/models/querysets/#field-lookups

## configurar entorno virtual dentro de la carpeta backend
## al clonar en casa por first time, cambiar en Pipfile la version de python 3.9

	pipenv shell
	pip install django==3.2
	pip install django-restframework
	pip install django-cors-headers
	pip install djoser

## configurar django project

	django-admin startproject crm_django

# configurar settings.py

Agregar variable para corsheaders, conexión al frontend



CORS_ALLOWED_ORIGINS = [
    'http://localhost:8080',
    ]

## Agregar apps


INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'rest_framework',
    'rest_framework.authtoken',
    'corsheaders',
    'djoser',
]

## Agregar middleware para el corsheaders:

    'django.contrib.sessions.middleware.SessionMiddleware',
    'corsheaders.middleware.CorsMiddleware',
    'django.middleware.common.CommonMiddleware',



## Agregar djoser a las urls:

from django.contrib import admin
from django.urls import path, include

urlpatterns = [
    path('admin/', admin.site.urls),
    
    path('api/v1', include('djoser.urls')),
    path('api/v1', include('djoser.urls.authtoken')),
]

## Inicializar base de datos:

	cd crm_django
	python manage.py makemigrations
	python manage.py migrate
	
## CREATE SUPERUSER
	python manage.py createsuperuser
# Lenvantar el servidor
	python manage.py runserver




## Parte 3

## Implement a loading bar
## Create a Django app for the leads
## Database model for leads
## Serializer for the leads
## View for the leads
## Set up the urls
## Create page for showing the leads
## Make it possible to adds leads

























front/configuraFront.txt
## CONFIGURA FRONT


## v-bind: mostrar todo lo de mi función
## v-slot

CODE WITH STEIN
TUTORIAL = https://www.youtube.com/watch?v=u2RJXTtCn2Y&t=1288s

Configurar .gitignore
https://www.freecodecamp.org/espanol/news/gitignore-explicado-que-es-y-como-agregar-a-tu-repositorio/



# Ignora archivos del sistema Mac 
.DS_store

# Ignora la carpeta node_modules
node_modules

# Ignora todos los archivos de texto
*.txt

# Ignora los archivos relacionados a API keys
.env

# Ignora archivos de configuración SASS
.sass-cache




# PUEDO OMITIR npm install -g @vue/cli si Ya lo hice alguna vez en la PC
npm install -g @vue/cli
vue create crm_vue
    manually select features

    chosse vue
    babel
    router
    Vuex
    CSS Pre-processors

    Chosse version 3.X

    Use history mode ruter YES
    Sass/SCSS dart-sass
    In dedicated config files N

    Si decidí crear nombre al proyecto
    cd proyecto

    ## Instalar dependencias
        npm install bulma
        npm install bulma-toast
        npm install axios

    ## Configurar axios en src/main.js
     import axios from 'axios'

     axios.defaults.baseURL = 'http://127.0.0.1:8000'
     createApp(App).use(store).use(router, axios).mount('#app')

    ## Configurar src/App.vue
    <template>
        <router-view/>
      </template>
      
      <style lang="scss">
      @import '../node_modules/bulma/';
      </style>
      
## Hacer posible hacer Registro de usuario desde Frontend
## Hacer posible hacer LogIn desde Frontend

## Hacer posible hacer Log out desde Frontend





back/crm_django/crm_django/settings.py
"""
Django settings for crm_django project.

Generated by 'django-admin startproject' using Django 3.2.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""

from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-+w-_cr(0-f1g2%520+m4bl1b@dfa9%iamah8m-%1m)a886zs)i'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


CORS_ALLOWED_ORIGINS = [
    'http://localhost:8080',
]

REST_FRAMEWORK = {
    'DEFAULT_AUTHENTICATION_CLASSES': (
        'rest_framework.authentication.TokenAuthentication',
    ),
    'DEFAULT_PERMISSION_CLASSES': (
        'rest_framework.permissions.IsAuthenticated',
    )
}


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'rest_framework',
    'rest_framework.authtoken',
    'corsheaders',
    'djoser',
    'lead.apps.LeadConfig'
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'corsheaders.middleware.CorsMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'crm_django.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'crm_django.wsgi.application'


# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': BASE_DIR / 'db.sqlite3',
    }
}


# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

LANGUAGE_CODE = 'es-mx'

TIME_ZONE = 'America/Mexico_City'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = '/static/'

# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
